src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -0,0 +1,73 @@
+import {useEffect, useState} from 'react';
+import type {ComponentType} from 'react';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import useNetwork from '@hooks/useNetwork';
+import {openPolicyAccountingPage} from '@libs/actions/PolicyConnections';
+import withPolicy from './withPolicy';
+import type {WithPolicyProps} from './withPolicy';
+
+type WithPolicyConnectionsProps = WithPolicyProps;
+
+/**
+ * Higher-order component that fetches the connections data and populates
+ * the corresponding field of the policy object if the field is empty. It then passes the policy object
+ * to the wrapped component.
+ *
+ * Use this HOC when you need the policy object with its connections field populated.
+ *
+ * Only the active policy gets the complete policy data upon app start that includes the connections data.
+ * For other policies, the connections data needs to be fetched when it's needed.
+ */
+function withPolicyConnections(WrappedComponent: ComponentType<WithPolicyConnectionsProps>) {
+    function WithPolicyConnections({policy, policyMembers, policyDraft, policyMembersDraft, route}: WithPolicyConnectionsProps) {
+        const {isOffline} = useNetwork();
+
+        // When the accounting feature is enabled, but the user hasn't connected to any accounting software,
+        // the connections data doesn't exist. We don't want to continually attempt to fetch non-existent data.
+        // This state helps us track whether a data fetch attempt has been made.
+        const [wasConnectionsDataFetched, setWasConnectionsDataFetched] = useState(false);
+
+        useEffect(() => {
+            // When the accounting feature is not enabled, or if the connections data already exists,
+            // there is no need to fetch the connections data.
+            if (wasConnectionsDataFetched || !policy?.areConnectionsEnabled || !!policy?.connections || !policy?.id) {
+                return;
+            }
+
+            openPolicyAccountingPage(policy.id);
+            setWasConnectionsDataFetched(true);
+        }, [policy, wasConnectionsDataFetched]);
+
+        if (!policy?.connections) {
+            if (isOffline) {
+                return (
+                    <FullPageOfflineBlockingView>
+                        <WrappedComponent
+                            policy={policy}
+                            policyMembers={policyMembers}
+                            policyDraft={policyDraft}
+                            policyMembersDraft={policyMembersDraft}
+                            route={route}
+                        />
+                    </FullPageOfflineBlockingView>
+                );
+            }
+
+            return null;
```

Comment:
I don't get why we return nothing if the `connections` are false, I try to use it on the workspace profile page but it shows a blank page. Also, I'm curious about what case the HOC will be used.

Reply:
I added some comments in [the new commit](https://github.com/Expensify/App/pull/39132/commits/23702781c6a90ee8786318d43323bb90c9b2326a). It explains the usage of the HOC and the context that I also added in that commit.

Reply:
> I don't get why we return nothing if the connections are false

@mollfpr, @aldo-expensify, should we return `LoadingPage` instead of `null`?

Reply:
We might be able to leverage this later. The only case I think we should return `null` is where we want to navigate the page if there are no `connections` to avoid rendering a component while animating the transition.

Reply:
I think we should should a spinner if `policy.isLoadingConnections` is true, and if `policy.isLoadingConnections` is false, hmmm 404? 🤷